static/js/narrow.js

```diff
@@ -234,11 +249,18 @@ exports.activate = function (raw_operators, opts) {
         ui.show_loading_more_messages_indicator();
     }
 
-    // Put the narrow operators in the URL fragment.
-    // Disabled when the URL fragment was the source
-    // of this narrow.
+    // Put the narrow operators in the URL fragment. Disabled when the URL
+    // fragment was the initiating source of this narrow.
     if (opts.change_hash) {
         hashchange.save_narrow(operators);
+    } else if (following_alias) {
+        // We want this to update the hash on page load if someone followed
+        // a link to a renamed stream. Since hashchange will refuse us right now
+        // (it's already tracking a page load), we process on a future tick.
+        // We set the "replace_state" parameter true to avoid polluting history.
+        setTimeout(function () {
+            hashchange.save_narrow(operators, true);
```

Comment:
hmm, ideally, we'd be able to do this by rewriting the narrow in the `hashchange.js` on-page-load code path, rather than loading the old narrow and then flickering it to the new state.